homeassistant/components/ps4/media_player.py

```diff
@@ -0,0 +1,373 @@
+        except TypeError:
+            _LOGGER.error(
+                "Could not find data in region: %s for PS ID: %s",
+                self._region, title_id)
+        finally:
+            self._media_title = app_name or name
+            self._source = self._media_title
+            self._media_image = art
+            self.update_list()
+
+    def update_list(self):
+        """Update Game List, Correct data if different."""
+        if self._media_content_id in self._games:
+            store = self._games[self._media_content_id]
+            if store != self._media_title:
+                self._games.pop(self._media_content_id)
+        if self._media_content_id not in self._games:
+            self.add_games(self._media_content_id, self._media_title)
+            self._games = self.load_games()
+        self._source_list = list(sorted(self._games.values()))
+
+    def load_games(self):
+        """Load games for sources."""
+        g_file = self._games_filename
+        try:
+            games = load_json(g_file)
+            return games
+        except FileNotFoundError:
+            games = {}
+            self.save_games(games)
+            self.load_games()
+
+    def save_games(self, games):
+        """Save games to file."""
+        g_file = self._games_filename
+        try:
+            save_json(g_file, games)
+        except OSError as error:
+            _LOGGER.error("Could not save game list, %s", error)
+
+    def add_games(self, title_id, app_name):
+        """Add games to list."""
+        games = self._games
+        if title_id is not None and title_id not in games:
+            game = {title_id: app_name}
+            games.update(game)
+            self.save_games(games)
+
+    def get_device_info(self, status):
+        """Return device info for registry."""
+        _sw_version = status['system-version']
+        _sw_version = _sw_version[1:4]
+        sw_version = _sw_version[0] + '.' + _sw_version[1:]
+        self._info = {
+            'name': status['host-name'],
+            'model': 'PlayStation 4',
+            'identifiers': {
+                (PS4_DOMAIN, status['host-id'])
+            },
+            'manufacturer': 'Sony Interactive Entertainment Inc.',
+            'sw_version': sw_version
+        }
+        self._unique_id = status['host-id']
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return self._info
+
+    @property
+    def unique_id(self):
+        """Return Unique ID for entity."""
+        return self._unique_id
+
+    @property
+    def entity_picture(self):
+        """Return picture."""
+        if self._state == STATE_OFF:
+            return None
+
+        image_hash = self.media_image_hash
+        if image_hash is not None:
+            return ENTITY_IMAGE_URL.format(
+                self.entity_id, self.access_token, image_hash)
+
+        if self._media_content_id is None:
+            return None
```

Comment:
Inconsistent return. Always return something explicitly in all cases if we return a value anywhere. The fourth and last case should also return something.

Reply:
So youre saying return a var with value None. Instead of just returning None.

Reply:
No, there are three if statements with return statements in this method. That makes it four cases that can happen. The fourth and last case should also have an explicit return statement and probably `return None`.